src/Storages/StorageReplicatedMergeTree.h

```diff
@@ -244,14 +243,15 @@ class StorageReplicatedMergeTree final : public MergeTreeData
     /** Remove a specific replica from zookeeper.
      * returns true if there are no replicas left
      */
-    static bool dropReplica(zkutil::ZooKeeperPtr zookeeper, const String & zookeeper_path, const String & replica,
+    static bool dropReplica(zkutil::ZooKeeperPtr zookeeper, const TableZnodeInfo & zookeeper_info,
                             LoggerPtr logger, MergeTreeSettingsPtr table_settings = nullptr, std::optional<bool> * has_metadata_out = nullptr);
 
-    bool dropReplica(const String & drop_zookeeper_path, const String & drop_replica, LoggerPtr logger);
+    bool dropReplica(const String & drop_replica, LoggerPtr logger);
 
     /// Removes table from ZooKeeper after the last replica was dropped
-    static bool removeTableNodesFromZooKeeper(zkutil::ZooKeeperPtr zookeeper, const String & zookeeper_path,
-                                              const zkutil::EphemeralNodeHolder::Ptr & metadata_drop_lock, LoggerPtr logger);
+    static bool removeTableNodesFromZooKeeper(
+        zkutil::ZooKeeperPtr zookeeper, const TableZnodeInfo & zookeeper_info2,
```

Comment:
Why not name the parameter `zookeeper_info`? It's a static method, and the name won't conflict with the class field.